### PR TITLE
Fix small bug in GasMeter calculation in creation context

### PR DESCRIPTION
**Pull request by hrkrshnn**
Old calculation did not ignore `optimize-runs` parameter, when creation code gas computations were
done. This would lead to expanding values such as `shl(180, 1)` inside constructor code, even for
the default value for `--optimize-runs`, i.e., 200.

Does it need a changelog? Would it only affect IR generation?

`--asm` doesn't have the same issue.

### CODE REVIEW

**Comment by chriseth** [comment]
It only affects IR

**Comment by chriseth** [comment]
Or actually maybe not? Can you check that this works as expected when optimizing ABIv2 when used in construction context wrt. when used in deployed code?

File: libyul/backends/evm/EVMMetrics.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ size_t GasMeter::instructionCosts(evmasm::Instruction _instruction) const
 
 size_t GasMeter::combineCosts(std::pair<size_t, size_t> _costs) const
 {
-	return _costs.first * m_runs + _costs.second;
+	return m_isCreation ? (_costs.first + _costs.second) : (_costs.first * m_runs + _costs.second);
```
Comment (chriseth) [inline libyul/backends/evm/EVMMetrics.cpp]: The same can be achieved by forcing `m_runs` to 1 in the constructor. Do you think that would be a better solution?

Comment (hrkrshnn) [reply libyul/backends/evm/EVMMetrics.cpp]: I forced `m_runs` to `1` for creation context.

**Review by chriseth — APPROVED** [review]

**Pull request merged**